src/user-guide/running-workflows/simulation-modes.rst

```diff
@@ -41,18 +41,56 @@ If :cylc:conf:`[runtime][<namespace>]execution time limit` and
 run length is computed by dividing the time limit by the speedup factor.
 
 
+Simulated Failure
+^^^^^^^^^^^^^^^^^
+
+By default all tasks succeed, sending all task outputs.
```

Comment:
That would include `failed` and `submit-failed`?

I'm guessing that by default simulation and dummy mode only send:

* submitted
* started
* succeeded

(and no custom outputs)

Reply:
From testing, it would appear that `itask.tdef.rtconfig['outputs'].values()` is empty (possibly unset by the simulation config?), so it looks like simulation mode only sets `succeeded` here.

Reply:
```diff
diff --git a/cylc/flow/task_pool.py b/cylc/flow/task_pool.py
index 92905becf..a71e8bc9e 100644
--- a/cylc/flow/task_pool.py
+++ b/cylc/flow/task_pool.py
@@ -1755,6 +1755,7 @@ class TaskPool:
 
     def sim_time_check(self, message_queue: 'Queue[TaskMsg]') -> bool:
         """Simulation mode: simulate task run times and set states."""
+        print('%')
         if not self.config.run_mode('simulation'):
             return False
         sim_task_state_changed = False
@@ -1783,6 +1784,8 @@ class TaskPool:
                 else:
                     # Simulate message outputs.
                     for msg in itask.tdef.rtconfig['outputs'].values():
+                        # this line doesn't seem to get called by default
+                        print('#', msg)
                         message_queue.put(
                             TaskMsg(job_d, now_str, 'DEBUG', msg)
                         )
```

Reply:
Counter-example :

```
[scheduler]
    allow implicit tasks = true

[scheduling]
    [[graph]]
        R1 = """
            foo:succeed => after_success
            foo:bar  => bar_follower
            foo:qux  => qux_follower
        """

[runtime]
    [[foo]]
        script = false
        [[[outputs]]]
            bar = 'It\'s not Irrelephant'
            qux = 'It\'s a Hippopotamus'
```

leads to the following:

```
INFO - [1/foo running job:01 flows:1] => succeeded
INFO - [1/after_success waiting(runahead) job:00 flows:1] => waiting
INFO - [1/after_success waiting job:00 flows:1] => waiting(queued)
INFO - [1/bar_follower waiting(queued) job:00 flows:1] => waiting
INFO - [1/qux_follower waiting(queued) job:00 flows:1] => waiting
INFO - [1/after_success waiting(queued) job:00 flows:1] => waiting
```
... then they run

I tried putting the print statement in as a `LOG.critical(msg)` where you had a print statement and got:

```
CRITICAL - It\'s not Irrelephant
CRITICAL - It\'s a Hippopotamus

```

Reply:
Custom outputs :vomiting_face:. That's really nasty for branched workflows where it could activate multiple branches for custom output switches, but not for succeeded/failed switches!

Counter counter example:

```
[scheduler]
    allow implicit tasks = true

[scheduling]
    [[graph]]
        R1 = """
            foo:started => started
            foo:submitted? => submitted
            foo:submit-failed? => submitted
            foo:expired => expired
            foo:succeed? => succeed
            foo:failed? => failed
            foo:custom  => custom
        """

[runtime]
    [[foo]]
        script = false
        [[[outputs]]]
            custom = custom
```

```
$ cylc vip . -N --mode=simulation 2>&1 | grep '=> waiting(queued)'
2023-10-16T14:10:12+01:00 INFO - [1/foo waiting job:00 flows:1] => waiting(queued)
2023-10-16T14:10:23+01:00 INFO - [1/custom waiting job:00 flows:1] => waiting(queued)
2023-10-16T14:10:23+01:00 INFO - [1/succeed waiting job:00 flows:1] => waiting(queued)
```

So in this case the only outputs produced were the succeeded and the custom but NOT:

* submitted
* submit-failed
* started
* expired
* failed

Reply:
OK. I'll go back and examine the logic in detail.

Reply:
From the above I don't think that's right either?

Reply:
I'm going to create a spin off issue for a possible bugfix, and focus on documenting the actual situation, else this is mission creep:

```suggestion
Tasks always complete all custom outputs.

By default they succeed, but they can be configured to fail using
`[]
```

Reply:
I've written up the situation as it currently is, and created [this issue](https://github.com/cylc/cylc-flow/issues/5774) - @oliver-sanders can you have a look and say whether you are happy with the current state of play. I suggest not resolving this discussion to make it easier to find using the link on the new issue.